src/app/pipes/center-truncate.pipe.ts

```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+    name: 'centerTruncate'
+})
+export class CenterTruncatePipe implements PipeTransform {
+    transform(value: string, maxLength: number): string {
```

Comment:
```
transform(value: string, maxLength: number): string {
    if (!value || value.length <= maxLength) return value;
    const halfLength = Math.floor((maxLength - 3) / 2); // Subtract 3 for "..."
    return value.slice(0, halfLength) + '...' + value.slice(-halfLength);
}
```

Reply:
Even better to define `function transform(value: string, maxLength: number = 16): string` to avoid passing `:[16]` at two places unless needed.